emissionsapi/utils.py

```diff
@@ -10,3 +10,34 @@ def bounding_box_to_wkt(lon1, lat1, lon2, lat2):
     """
     return f'POLYGON(({lon1} {lat1},{lon1} {lat2},{lon2} {lat2},'\
            f'{lon2} {lat1},{lon1} {lat1}))'
+
+
+def polygon_to_wkt(polygon):
+    """
+    Takes a point list and converts them to a wkt-polygon.
+    The list of poinst has to be provided in the form of
+    (lon1, lat1, lon2, lat2, ...).
+    Therefor, the number of elements in the list has to be even.
+    If the polygon is not closed (the first lon/lat pair does not match
+    the last lon/lat pair), the polygon will be closed here.
+    """
+    # check if element number is even
+    if len(polygon) % 2 != 0:
+        raise ValueError('Incorrect number of elements (has to be even).')
+
+    # check if polygon is closed
+    if polygon[0] != polygon[-2] or polygon[1] != polygon[-1]:
+        # close polygon by adding the first lon/lat pair at the end of the list
+        polygon.extend(polygon[0:2])
+
+    # check if we have at least 3 (+1 to close the polygon) coordinate points
+    if len(polygon) < 8:
+        raise ValueError('Incorrect number of elements (we need at least 8).')
+
+    # create list with x-y points as strings
+    points = []
+    for index in range(0, len(polygon), 2):
```

Comment:
Not sure if this isn't easier to read:

```
    points = [f'{lon} {lat}' for lon, lat in zip(polygon[::2], polygon[1::2])]
```

Reply:
Hehe, actually @shaardie suggested nearly exactly the same to me while I was working on that part of the code. But I decided to leave it this way since for me (as a python noob) this is muche easier to read.

Reply:
I feared indices would be more comfortable for C fanatics… 🙀
Out of interest, what makes this harder to read? The `zip(…)` function? Or slicing the list using `[::2]`?

Reply:
I am not yet familiar with python list - so the list operations in combination with multiple functionalities in one line is not easy to read for me. So I tried to follow 'Readability counts' and write it in more than one line...